features/poller/client-rpc/src/test/java/org/opennms/netmgt/poller/client/rpc/PollerRequestDTOTest.java

```diff
@@ -50,87 +56,128 @@ public PollerRequestDTOTest(PollerRequestDTO sampleObject, String sampleXml) {
+                "         <retries>0</retries>\n" +
+                "         <securityLevel>1</securityLevel>\n" +
+                "         <securityName>opennmsUser</securityName>\n" +
+                "         <timeout>3000</timeout>\n" +
+                "         <version>1</version>\n" +
+                "         <versionAsString>v1</versionAsString>\n" +
+                "         <writeCommunity>private</writeCommunity>\n" +
+                "      </snmpAgentConfig>\n" +
+                "   </attribute>\n" +
                 "</poller-request>"
             }
         });
     }
-    public static PollerRequestDTO getPollerRequestWithString() throws UnknownHostException {
+    public static PollerRequestDTO getPollerRequestWithString() throws Exception {
         PollerRequestDTO dto = new PollerRequestDTO();
         dto.setLocation("MINION");
         dto.setClassName("org.opennms.netmgt.poller.monitors.IcmpMonitor");
         dto.setAddress(InetAddress.getByName("127.0.0.1"));
-        dto.addAttribute("port", "18980");
+        dto.addAttribute("port", PollerParameter.simple("18980"));
         return dto;
     }
 
-    public static PollerRequestDTO getPollerRequestWithStringAsObject() throws UnknownHostException {
+    public static PollerRequestDTO getPollerRequestWithAgentConfig() throws Exception {
         PollerRequestDTO dto = new PollerRequestDTO();
         dto.setLocation("MINION");
         dto.setClassName("org.opennms.netmgt.poller.monitors.IcmpMonitor");
         dto.setAddress(InetAddress.getByName("127.0.0.1"));
-        // Casting the String to an Object will use a different code path
-        dto.addAttribute("port", (Object)"18980");
+        dto.addAttribute("agent", PollerParameter.complex(createSnmpAgentConfig()));
         return dto;
     }
 
-    public static PollerRequestDTO getPollerRequestWithObject() throws UnknownHostException {
-        PollerRequestDTO dto = new PollerRequestDTO();
-        dto.setLocation("MINION");
-        dto.setClassName("org.opennms.netmgt.poller.monitors.IcmpMonitor");
-        dto.setAddress(InetAddress.getByName("127.0.0.1"));
-        dto.addAttribute("nested", new PollerAttributeDTO("x", "y"));
-        return dto;
-    }
+    private static Element createSnmpAgentConfig() throws Exception {
+        final DocumentBuilder documentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+        final Document document = documentBuilder.newDocument();
 
-    public static PollerRequestDTO getPollerRequestWithAgentConfig() throws UnknownHostException {
-        PollerRequestDTO dto = new PollerRequestDTO();
-        dto.setLocation("MINION");
-        dto.setClassName("org.opennms.netmgt.poller.monitors.IcmpMonitor");
-        dto.setAddress(InetAddress.getByName("127.0.0.1"));
-        dto.addAttribute("agent", new SnmpAgentConfig());
-        return dto;
+        final Element rootElement = document.createElementNS(null, "snmpAgentConfig");
```

Comment:
Can we actually marshal the `SnmpAgentConfig` object instead of building the DOM manually?

Reply:
With the latest changes, I think we can.